frontend/src/actionCreators/permitActionCreator.js

```diff
@@ -0,0 +1,128 @@
+    .then((response) => {
+      dispatch(success(reducerTypes.GET_PERMITS));
+      dispatch(permitActions.storePermits(response.data));
+      dispatch(hideLoading("modal"));
+    })
+    .catch(() => {
+      notification.error({
+        message: String.ERROR,
+        duration: 10,
+      });
+      dispatch(error(reducerTypes.GET_PERMITS));
+      dispatch(hideLoading("modal"));
+    });
+};
+
+export const updatePermit = (permit_guid, payload) => (dispatch) => {
+  dispatch(request(reducerTypes.UPDATE_PERMIT));
+  dispatch(showLoading());
+  return axios
+    .put(`${ENVIRONMENT.apiUrl}${API.PERMIT()}/${permit_guid}`, payload, createRequestHeader())
+    .then((response) => {
+      notification.success({
+        message: `Successfully updated permit`,
+        duration: 10,
+      });
+      dispatch(success(reducerTypes.UPDATE_PERMIT));
+      dispatch(hideLoading());
+      return response;
+    })
+    .catch((err) => {
+      notification.error({
+        message: err.response ? err.response.data.error.message : String.ERROR,
+        duration: 10,
+      });
+      dispatch(error(reducerTypes.UPDATE_PERMIT));
+      dispatch(hideLoading());
+    });
+};
+
+export const createPermitAmendment = (permitGuid, payload) => (dispatch) => {
+  dispatch(request(reducerTypes.CREATE_PERMIT_AMENDMENT));
+  dispatch(showLoading("modal"));
+  return axios
+    .post(
+      `${ENVIRONMENT.apiUrl}${API.PERMITAMENDMENTS(permitGuid)}`,
+      payload,
+      createRequestHeader()
+    )
+    .then((response) => {
+      notification.success({ message: "Successfully created a new permit", duration: 10 });
+      dispatch(success(reducerTypes.CREATE_PERMIT_AMENDMENT));
+      dispatch(hideLoading("modal"));
+      return response;
+    })
+    .catch((err) => {
+      notification.error({
+        message: err.response ? err.response.data.error.message : String.ERROR,
+        duration: 10,
+      });
+      dispatch(error(reducerTypes.CREATE_PERMIT_AMENDMENT));
+      dispatch(hideLoading("modal"));
+    });
+};
+
+export const updatePermitAmendment = (permitAmdendmentGuid, payload) => (dispatch) => {
+  dispatch(request(reducerTypes.UPDATE_PERMIT_AMENDMENT));
+  dispatch(showLoading());
+  return axios
+    .put(
+      `${ENVIRONMENT.apiUrl}${API.PERMITAMENDMENT(permitAmdendmentGuid)}`,
+      payload,
+      createRequestHeader()
+    )
+    .then((response) => {
+      notification.success({
+        message: `Successfully updated permit amendment`,
+        duration: 10,
+      });
+      dispatch(success(reducerTypes.UPDATE_PERMIT_AMENDMENT));
+      dispatch(hideLoading());
+      return response;
+    })
+    .catch((err) => {
+      notification.error({
+        message: err.response ? err.response.data.error.message : String.ERROR,
+        duration: 10,
+      });
+      dispatch(error(reducerTypes.UPDATE_PERMIT_AMENDMENT));
+      dispatch(hideLoading());
+    });
+};
```

Comment:
I love that you broke permits out into their own Redux file structure, great job.